### DISCUSSION TITLE
How to set wave input on a boundary

### DISCUSSION THREAD

**Discussion by 675997215** [NOAA-EMC/WW3#706] [General]
I just started using W recently, and some problems bothered me. I want to build a wave tank in WW3 (in Cartesian coordinates). At one end of the tank, there is an irregular wave input that can be set, similar to the wave maker. I tried to set the initial conditions of the wave field, and I found that it was not ideal. Now I'm thinking about using nested meshes. Is there any way to realize the generation of waves similar to wave maker.

**Comment by aliabdolali** [comment]
@675997215 
Yes, this is fairly easy. You can prepare boundary conditions in either ascii or netcdf and use ww3_bound (for ascii) and ww3_bounc (for netcdf). The spectrum should represent your irregular wave conditions. The directional spectral density should follow WW3 format. Don't forget to define the open boundary nodes in ww3_grid.inp(nml).

**Reply by 675997215** [reply]
Thank you very much for your help. One thing I want to confirm is whether the wave spectrum of the boundary needs to be input only once or every time of the simulation. Another problem is that I need to set the time step in the ww3_grid.inp file. I use Cartesian coordinates and DXY is 3m. In this way, the calculated DTMAX is 0.45s, but an error will be reported when running WW3. I found that the minimum time step is 1s. How to set the time step in the small-scale grid. (single grid is 3m * 3m)


The following is the calculation method given in the user manual

! Tcfl = DXY / (G / (FREQ1*4*Pi) ) with the constants Pi=3,14 and G=9.
! DTXY ~= 90% Tcfl
! DTMAX ~= 3 * DTXY (maximum global time step limit)
!
! * The refraction time step depends on how strong can be the current ve
! on your grid :
! DTKTH ~= DTMAX / 2 ! in case of no or light current velocities
! DTKTH ~= DTMAX / 10 ! in case of strong current velocities
!
! * The source terms time step is usually defined between 5s and 60s.
! A common value is 10s.
! DTMIN ~= 10

**Reply by aliabdolali** [reply]
The boundary condition is a time series, if it is stationary, you can have one/two time steps, and the model extrapolates/interpolates between them. I usually generate one for the beginning of the simulation and one for the end time of the simulation. Therefore the model interpolates between them.

You can add sec1 to your switch and use smaller time steps than 1s.

**Reply by 675997215** [reply]
hi, aliabdolali

In my laboratory tests, unidirectional waves are generated, but there is an angle between the wind field and the wave direction, so the propagation in the Y direction should be considered in the numerical simulation. However, there is a big gap between the calculated results and the experimental results.

I set up a wave spectrum on the boundary to generate waves (similar to a wave maker). When I close FLCY in file ww3_grid.inp and WW3 runs for a period of time, the whole wave field shows a stable significant wave height (slightly decreased due to wave dissipation). But when I turn on FLCY, the significant wave height decreases significantly along the propagation direction, which is not what I expected (the energy loss cannot be so large). What is the reason for this?
